### ISSUE TITLE
Discount detail report functionality - enhancement?

### ISSUE THREAD

**Issue by esbyrt** [opensourcepos/opensourcepos#1736]
OSPOS version is: 3.2.0
OSPOS git commit hash is: 4f5ad57
PHP version is: (e.g. 5.5, 5.6, 7.0, 7.1) 7.1
MySQL or MariaDB version is: (e.g. MySQL 5.5, MySQL 5.6, MySQL 5.7, MariaDB 10.0, MariaDB 10.1, MariaDB 10.2) 10.0 .32
OS and version is: (e.g. CentOS 6.9, Ubuntu 16.4, Windows 10) Windows 10/Windows 8
WebServer is: (e.g. Apache 2.2, Apache 2.4, Nginx 1.12, Nginx 1.13) Apache 2.4.27

### Issue / Question / Bug

Before submitting an issue please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [x] I'm reporting an issue of an unmodified OSPOS installation
- [x] I checked open and closed issues database and no similar issue was already discussed (please make sure you searched!)
- [x ] I read the README, WHATS_NEW and UPGRADE
- [x] I read the FAQ (https://github.com/jekkos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP has php-gd, php-intl, sockets and etc. installed)
- [x] I read the wiki
- [] I ran any database upgrade scripts (e.g. database/2.4_to_3.0.sql), and migrating function
- [x] I'm aware the latest master could be a development version and therefore not stable
- [x] I know the version of OSPOS and git commit hash (check the footer of your OSPOS), the name and version of OS, Web server, PHP and MySQL and will add them to my issue report

### Expected behaviour
Run the Detailed Discounts report for 10% and get only the sale items that were discounted 10%

### Actual behaviour
Report returns all items from a sale that has one item discounted 10%

### Steps to reproduce the issue
Enter a sale with several different discount amounts and run the report
![discount report](https://user-images.githubusercontent.com/28318020/33391172-781a8674-d4f5-11e7-99a1-6e221565b845.JPG)
I need to calculate the actual dollar amount of discounts so this report doesn't work as is. If I could pull a 10% discount report that showed only the line items that were discounted 10% I could calculate the dollar value. In the picture I uploaded the first item that shows as $22 was discounted 60% so $33 dollars off the price. The total at the bottom of the report includes all of the line items, not just the 10% ones.

**Comment by SteveIreland** [comment]
Sounds like a job for Reportico! (or something similar).

**Comment by daN4cat** [comment]
Yes. I'm waiting for Reportico guy to complete the rearch work. He shouldn't be too far.

**Comment by esbyrt** [comment]
Thanks for pointing me at Reportico! I have a learning curve there but it sure looks like it will do what I need.

**Comment by daN4cat** [comment]
One day it will be part of OSPOS. Sometime in 2018....

**Comment by odiea** [comment]
In 3.2.0 on this report the tax and total columns are not correct in the detailed part of the report.
Tax column should say Total and Total column should say Tax.

**Comment by daN4cat** [comment]
@odiea thanks for the issue report, let me check and fix.

**Comment by daN4cat** [comment]
Fixed and in master

**Comment by odiea** [comment]
@SteveIreland has requested a new commit. I will wait till his is commited into OSPOS.
Reports Specific Discount enhancement.
1. I have added a discount dropdown to Register in 5% increments. To show proper discounts in Discount Report.
2. I have updated Specific Discount Report to show just the Discount % and amount for that % for the Sale.
If there is any interest in this let me know.

**Comment by daN4cat** [comment]
Send the changes to the report that I will look into that and add to close this issue.

**Comment by daN4cat** [comment]
Thanks for the help @odiea

**Comment by odiea** [comment]
This is just the Specific Discount Report with no changes to the current 10% drop down box selection.
[Specific_discount.zip](https://github.com/opensourcepos/opensourcepos/files/1595883/Specific_discount.zip)

**Comment by daN4cat** [comment]
@odiea I'm not sure about this because the discount is in brackets, where your options looks for precisely a specific discount.

**Comment by odiea** [comment]
Yes it does look for a specific % and shows only that discount on the report. So if someone used something other than increments of 10 for a discount it will not show. This was the only way I could get the % to show on the Report without other %'s in the Sale showing. So it may not be useful to some.

**Comment by odiea** [comment]
This image is a Sale that includes 4 10% and 1 20% discount. POS id is 2194.
![2018-01-01 11_35_13-](https://user-images.githubusercontent.com/10962177/34470130-0da5d2aa-eee8-11e7-9d8e-8c5ff8b773da.png)

**Comment by odiea** [comment]
Here is another option. I changed the summary data back to the original code and I got the following. It only totals the 10% items but also now shows the 20% item.
![2018-01-01 11_52_27-wedidit _ powered by ospos 3 2 0](https://user-images.githubusercontent.com/10962177/34470240-a0848be6-eeea-11e7-94cb-8c604a3124df.png)

**Comment by daN4cat** [comment]
It requires changes to the discount selection form in order to select anyone you want. More work to do...

**Comment by daN4cat** [comment]
BTW, @odiea, your family name is Scottish and your clan tartan is the following: https://www.kinlochanderson.com/tartan/agnew

You can wear only that tartan and you should investigate which Scottish clan is on your side and which is not :-)

**Comment by odiea** [comment]
@daN4cat Thanks for the info. My dna test showed I was more irish. My mom was 100%. Dad didn't know so he said he was a Heinz 57.

**Comment by daN4cat** [comment]
Could be, but for sure your family name is Scottish.
You can even find it in Northern Ireland but due to the Scottish migration there during the Plantation of Ireland: https://en.wikipedia.org/wiki/Plantations_of_Ireland

**Comment by daN4cat** [comment]
More about your clan: https://en.wikipedia.org/wiki/Clan_Agnew

**Comment by daN4cat** [comment]
Now you need to buy a cap with your clan tartan and wear it :-) And maybe travel to Scotland once.... whisky is good.

**Comment by odiea** [comment]
I changed my avatar to the proper kilt I think.

**Comment by daN4cat** [comment]
Tartan

**Comment by daN4cat** [comment]
(I will remove this personal thread to avoid polluting the technical discussion)

**Comment by odiea** [comment]
I guess the only other solution I can come up with is to change from the dropdown % list to a user entry.

**Comment by esbyrt** [comment]
Hi guys. I have been following this thread. I haven't had time to figure out reportico to solve this one for me yet so... For my purposes I just need a dollar figure of total discounts for the month. So a drop selection of "All" would be great for me as long as it excluded sales that had no discount at all. Also a column showing the dollar discount amount as well as percentage. I'm not sure that's possible and I know I can get it from Reportico once I figure it out but I'm sure other users would appreciate that functionality as well.
